contrib/tools/meta.md/purification.md

```diff
@@ -201,17 +203,20 @@ fmod PURIFICATION is
    protecting BREAK-EQATOMS .
    protecting CTERM-SET .
 
-    var Q : Qid . var TA : TruthAtom . vars EqC EqC' : EqConj . var QFF : QFForm .
+    vars Q Q' : Qid . vars F F' : Form .
     vars ME ME' : ModuleExpression . vars M M' : Module . var MDS : ModuleDeclSet .
-    vars FV : Variable . vars T T' T1 T2 : Term . var T? : [Term] .
+    vars FV : Variable . vars T T' T1 T2 : Term . var T? : [Term] . var CT : CTerm .
     vars NeTL NeTL' : NeTermList . vars TL TL' : TermList . vars TL? TL?' : [TermList] .
 
-    op _in_ : EqConj Module -> Bool .
-    ---------------------------------
-    eq TA            in M = true .
-    eq (EqC /\ EqC') in M = (EqC in M) and (EqC' in M) .
-    eq (T ?= T')     in M = wellFormed(M, T) and wellFormed(M, T') .
-    eq (T != T')     in M = wellFormed(M, T) and wellFormed(M, T') .
+    op _in_ : Form Module -> Bool .
+    -------------------------------
+    eq tt        in M = true .
+    eq ff        in M = true .
+    eq (~ F)     in M = F in M .
+   ceq (F /\ F') in M = (F in M) and (F' in M) if F =/= tt /\ F' =/= tt .
+   ceq (F \/ F') in M = (F in M) and (F' in M) if F =/= ff /\ F' =/= ff .
+    eq (T ?= T') in M = wellFormed(M, T) and wellFormed(M, T') .
+    eq (T != T') in M = wellFormed(M, T) and wellFormed(M, T') .
```

Comment:
I think also here the side conditions are sufficient?

Reply:
This works too.